### PR TITLE
GEN-1015: Feature/bazooka/chat

### PR DESCRIPTION
**Pull request by sladan-hedvig**
## [APP-XXX]

- Moved giraffe to octopus
- Updated logic for notification toast

## Checklist

- [ ] Dark/light mode verification
- [ ] Landscape verification
- [ ] iPad verification
- [ ] iPod/iPhone 12 mini/iPhone SE verification

### CODE REVIEW

File: Projects/App/Sources/Screens/ChatScreen/Models/Message.swift [hunk 1]
```diff
@@ -335,11 +275,11 @@ func hash(into hasher: inout Hasher) { hasher.combine(globalId) }
             type = .text
         }
 
-        fromMyself = message.header.fromMyself
-        statusMessage = message.header.statusMessage
-
-        let timeStampInt = Int(message.header.timeStamp) ?? 0
-        timeStamp = TimeInterval(timeStampInt / 1000)
+        fromMyself = message.sender == .member
+        statusMessage = nil  //message.header.statusMessage
```
Comment (julia-andersson) [inline Projects/App/Sources/Screens/ChatScreen/Models/Message.swift:279]: remove comment

File: Projects/App/Sources/Screens/ChatScreen/Reusables/Message+Reusable.swift [hunk 2]
```diff
@@ -369,6 +398,118 @@ extension Message: Reusable {
...
+                        imageView.snp.makeConstraints { make in
+                            make.leading.equalToSuperview()
+                            make.trailing.equalToSuperview()
+                            make.top.equalToSuperview().offset(message.shouldShowTimeStamp ? 12 : 12)
+                            make.width.equalTo(imageView.snp.height)
+                        }
+
+                        imageView.layer.cornerRadius = 20
+                        imageView.clipsToBounds = true
+                        //title //subtitle
```
Comment (julia-andersson) [inline Projects/App/Sources/Screens/ChatScreen/Reusables/Message+Reusable.swift:429]: why is this here?

File: Projects/App/Sources/Screens/ChatScreen/Viewables/AttachGIFPane.swift [hunk 3]
```diff
@@ -11,6 +11,7 @@ struct AttachGIFPane {
     let isOpenSignal: ReadWriteSignal<Bool>
     let chatState: ChatState
     @Inject var giraffe: hGiraffe
```
Comment (julia-andersson) [inline Projects/App/Sources/Screens/ChatScreen/Viewables/AttachGIFPane.swift:13]: remove giraffe

File: Projects/App/Sources/Screens/ChatScreen/Viewables/ChatInput.swift [hunk 4]
```diff
@@ -188,9 +192,10 @@ extension ChatInput: Viewable {
 
         bag += chatState.currentMessageSignal.animated(style: SpringAnimationStyle.lightBounce()) { message in
             guard let message = message else {
-                inputBar.alpha = 0
-                singleSelectContainer.alpha = 0
-                audioContainer.alpha = 0
+                //                inputBar.alpha = 0
```
Comment (julia-andersson) [inline Projects/App/Sources/Screens/ChatScreen/Viewables/ChatInput.swift:195]: remove comment

**Review by julia-andersson — CHANGES_REQUESTED** [review]
Nice work!!

I don't get any time stamps in the chat anymore?

File: Projects/App/Sources/Screens/ChatScreen/Reusables/Message+Reusable.swift (hunk 2 above)
Comment (sladan-hedvig) [reply Projects/App/Sources/Screens/ChatScreen/Reusables/Message+Reusable.swift:429]: Thats something we did during maker days. Not a big thing :) but looks nice

**Comment by sladan-hedvig** [comment]
Should be fine now 👍

**Comment by sladan-hedvig** [comment]
@julia-andersson elvin did add initial message and it looks great!
